lib/spack/spack/installer.py

```diff
@@ -1879,6 +1883,9 @@ def __init__(self, pkg, install_args):
         # whether to enable echoing of build output initially or not
         self.verbose = install_args.get("verbose", False)
 
+        # whether installation was explicitly requested by the user
+        self.explicit = install_args.get("is_root", False) and install_args.get("explicit", False)
```

Comment:
```suggestion
        self.explicit = install_args.get("is_root", False) and install_args.get("explicit", True)
```

should duplicate `task.explicit`